src/LessMsi.Core/Msi/Wixtracts.cs

```diff
@@ -328,7 +328,8 @@ public static void ExtractFiles(Path msi, string outputDir, MsiFile[] filesToExt
                             LessIO.Path destName = LessIO.Path.Combine(targetDirectoryForFile, entry.LongFileName);
 					        if (FileSystem.Exists(destName))
 					        {
-						        Debug.Fail(string.Format("output file '{0}' already exists. We'll make it unique, but this is probably a strange msi or a bug in this program.", destName));
+						        Debug.WriteLine(string.Format("output file '{0}' already exists. We'll make it unique, but this is probably a strange msi or a bug in this program.", destName));

```

Comment:
It is a shame to lose this assertion here as it is a red flag for many issues I've seen with lessmsi over the years. In fact, even the Debug.WriteLine is invalid and misleading in the case of the deliberate overwrite mode. 

Could we feasibly add a new optional argument to `ExtractFiles` with the `ExtractionMode` and then, right here not assert if there is a duplicate?

Reply:
Actually if you add the `ExtractionMode` arg to the method here, maybe you can just skip writing duplicate files in the first place, and then there will be no need to delete them later. Is that feasible?

Reply:
Hello @activescott.

I have amended the code according to your comments.
Please let me know if you have any other comments.

Thank you.